lib/core/test/bench/db/Main.hs

```diff
@@ -551,31 +565,66 @@ mkInputs prefix n =
   where
     lbl = show prefix <> "in"
 
-mkOutputs :: Int -> Int -> [TxOut]
-mkOutputs prefix n =
+-- | Returns Output transactions that effectively have no "TokenBundles", but a
+-- single Coin.
+mkOutputsCoin :: Int -> Int -> [TxOut]
+mkOutputsCoin prefix n =
     [ force
         (TxOut (mkAddress prefix i) (TokenBundle.fromCoin $ Coin 1))
     | !i <- [1..n]
     ]
 
+-- | Returns Output transactions that also contains non-empty multi asset
+-- tokens.
+mkOutputsToken :: Int -> Int -> Int -> Int -> [TxOut]
+mkOutputsToken assetCount tokenQuantity prefix n =
+    [ force (createTrans i)
+    | !i <- [1..n]
+    ]
+  where
+    createTrans i =
+        let tokenPolicyIdHexStringLength = 56
+            tokens = [
+                let tokenPolicyId =
+                        fromRight (error "Couldn't decode tokenPolicyId")
+                        . fromText
+                        . T.pack
+                        . take tokenPolicyIdHexStringLength
+                        . join
+                        . replicate tokenPolicyIdHexStringLength
+                        . show
+                        $ ac
+                    tokenName = fromRight (error "Couldn't decode tokenName")
+                        . fromText
+                        . T.pack
+                        . show
+                        $ ac
+                in (TokenMap.AssetId tokenPolicyId tokenName
+                    , TokenQuantity $ fromIntegral tokenQuantity)
+                    | !ac <- [1..assetCount] ]
+        in (TxOut (mkAddress prefix i)
+            (TokenBundle.TokenBundle (Coin 1) (TokenMap.fromFlatList tokens)))
```

Comment:
**Suggestion**: remove most of the nesting here, as it makes it rather hard to read!

(In particular, It's hard to connect the `[` with the `]` in the list comprehension, as they start and end on different columns, and are separated by several lines.)

Since things like `tokens` are in fact constants, they can be defined at the top-level of the `where` clause:

```suggestion
    [ force (mkTxOut i)
    | !i <- [1..n]
    ]
  where
    mkTxOut i = TxOut
        (mkAddress prefix i)
        (TokenBundle.TokenBundle (Coin 1) (TokenMap.fromFlatList tokens))
    mkTokenName = fromRight (error "Couldn't decode tokenName")
        . fromText . T.pack . show
    mkTokenPolicyId = fromRight (error "Couldn't decode tokenPolicyId")
        . fromText
        . T.pack
        . take tokenPolicyIdHexStringLength
        . join
        . replicate tokenPolicyIdHexStringLength
        . show
    tokenPolicyIdHexStringLength = 56
    tokens =
        [ ( TokenMap.AssetId (mkTokenPolicyId ac) (mkTokenName ac)
          , TokenQuantity $ fromIntegral tokenQuantity
          )
        | !ac <- [1 .. assetCount]
        ]
```